GiwazipClient/Views/PostImageView.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  PostImageView.swift
+//  GiwazipClient
+//
+//  Created by creohwan on 2023/01/25.
+//
+
+import UIKit
+
+import SnapKit
+
+class PostImageView: BaseViewController, UIGestureRecognizerDelegate {
+    
+    // MARK: - Property
+    private let scrollView = UIScrollView()
+    private let pinch = UIPinchGestureRecognizer()
+    
+    // MARK: - View
+    
+    var postImage: UIImageView = {
+        $0.image = UIImage(named: "cat")
+        $0.isUserInteractionEnabled = true
+        return $0
+    }(UIImageView())
+    
+    // MARK: - Method
+    
+    override func attribute() {
+        super.attribute()
+        setupScrollView()
+        setupImageView()
+        setupPintch()
+    }
+    
+    override func layout() {
+        view.addSubview(scrollView)
+        scrollView.addSubview(postImage)
+    }
+    
+    private func setupScrollView() {
+        scrollView.frame = view.bounds
+        scrollView.delegate = self
+        scrollView.zoomScale = 1.0
+        scrollView.minimumZoomScale = 1.0
+        scrollView.maximumZoomScale = 3.0
+        scrollView.contentInsetAdjustmentBehavior = .never
+        scrollView.showsVerticalScrollIndicator = false
+        scrollView.showsHorizontalScrollIndicator = false
+    }
+    
+    private func setupImageView() {
+        postImage.frame = scrollView.bounds
+        postImage.contentMode = .scaleAspectFit
+    }
+    
+    private func setupPintch() {
+        let pinch = UIPinchGestureRecognizer(target: self, action: #selector(controlPinch))
+        view.addGestureRecognizer(pinch)
+    }
+    
+    @objc func controlPinch() {
+        postImage.transform = postImage.transform.scaledBy(x: pinch.scale, y: pinch.scale)
+        pinch.scale = 1
+    }
+
+}
+
+extension PostImageView: UIScrollViewDelegate {
+    func viewForZooming(in scrollView: UIScrollView) -> UIView? {
+         return self.postImage
+     }
+    
+    func scrollViewDidZoom(_ scrollView: UIScrollView) {
+        if scrollView.zoomScale > 1 {
+            guard let image = postImage.image else { return }
+            guard let zoomView = viewForZooming(in: scrollView) else { return }
+            
+            let widthRatio = zoomView.frame.width / image.size.width
+            let heightRatio = zoomView.frame.height / image.size.height
+            let ratio = widthRatio < heightRatio ? widthRatio : heightRatio
+            
+            let newWidth = image.size.width * ratio
+            let newHeight = image.size.height * ratio
+            
+            let left = 0.5 * (newWidth * scrollView.zoomScale > zoomView.frame.width ?
+                              (newWidth - zoomView.frame.width) : (scrollView.frame.width - scrollView.contentSize.width))
+            let top = 0.5 * (newHeight * scrollView.zoomScale > zoomView.frame.height ? (newHeight - zoomView.frame.height) : (scrollView.frame.height - scrollView.contentSize.height))
```

Comment:
p5;
코드 리딩에 약간의 혼란이 있어서, 아래처럼 한번 더 괄호로 묶어주시는 건 어떨까요?
```suggestion
            let left = 0.5 * ((newWidth * scrollView.zoomScale > zoomView.frame.width) ?
                              (newWidth - zoomView.frame.width) : (scrollView.frame.width - scrollView.contentSize.width))
            let top = 0.5 * ((newHeight * scrollView.zoomScale > zoomView.frame.height) ? (newHeight - zoomView.frame.height) : (scrollView.frame.height - scrollView.contentSize.height))
```

Reply:
> p5; 코드 리딩에 약간의 혼란이 있어서, 아래처럼 한번 더 괄호로 묶어주시는 건 어떨까요?
> 
> 추가로 ScrollViewDidZoom 함수의 실행 과정을 한번 설명해주시면 좋을 것 같습니다. 코드 리딩으로 파악한다고 파악하긴 했지만, 대략적인 예측이고 정확한 동작은 코드 작성자 분이 더 잘 알 것 같아서요.

333cd3b 반영했습니다~!

Reply:
> > p5; 코드 리딩에 약간의 혼란이 있어서, 아래처럼 한번 더 괄호로 묶어주시는 건 어떨까요?
> > 추가로 ScrollViewDidZoom 함수의 실행 과정을 한번 설명해주시면 좋을 것 같습니다. 코드 리딩으로 파악한다고 파악하긴 했지만, 대략적인 예측이고 정확한 동작은 코드 작성자 분이 더 잘 알 것 같아서요.
> 
> [333cd3b](https://github.com/Samsamhada/Giwazip-Client-App/commit/333cd3b7ef8f80711c8eb7706b20dffc85f39d13) 반영했습니다~!

확인했습니다. 리뷰 리졸브합니다~